usethis 1.3.0

jennybc opened the pull request:
Still a WIP but >80% done. Happy to get feedback now or I can ping again when I consider it done.

@jimhester @hadley what do you think of the devtools backstory? People do ask about this, so I thought this might be a good opportunity to explain it all.

jennybc commented:
I wish the headings of sections were somehow automatically linked, the way they are on, say, typical GitHub pages. It is easy to imagine scenarios where we want to hyperlink back into these posts and to link to specific sections. Is that hard to do?

batpigandme commented:
Re. the links, I'll look into it today: I realised that's one of the changes in the switch from `<div>` to `<section>` in the latest update for rendering (pandoc?). So, for example, I need to go in and add sidebar nav for the past couple posts.

jimhester commented on content/articles/2018-02-usethis-1-3-0.Rmd:
```diff
@@ -0,0 +1,106 @@
+
+[devtools](https://github.com/r-lib/devtools#readme) is an R package that helps you create, develop, and install R packages. By 2017, it was clear that devtools would be easier to maintain if it was split into a few smaller and more focused packages. usethis is the first of these to be released publicly on CRAN. usethis is the home for functions related to package and project setup.
+
+devtools will continue to exist as a meta-package, exposing functionality from lower-level, domain-specific packages. In the long-run, a useR should attach devtools, via `library("devtools")`, and then call devtools functions in their daily workflow, as usual. There are two reasons you might need to know about our internal refactoring:
```
The third reason is possibly what repository to open an issue in.

jimhester commented on content/articles/2018-02-usethis-1-3-0.Rmd:
```diff
@@ -0,0 +1,106 @@
+
+devtools will continue to exist as a meta-package, exposing functionality from lower-level, domain-specific packages. In the long-run, a useR should attach devtools, via `library("devtools")`, and then call devtools functions in their daily workflow, as usual. There are two reasons you might need to know about our internal refactoring:
+
+  * The Awkward Adolescent Phase. It is impossible to choreograph the release of a suite of packages on CRAN. Therefore, there may be periods when functionality exists in, e.g., usethis, that is not yet exposed by devtools. During such times, you would access those specific functions directly through usethis. This implies you should attach devtools first and then any of its constituent packages,  ...
```
This should really only until the next devtools release and never happen again afterwards rather than being something that could crop up randomly in the future.

batpigandme replied:
I think the order of the words here with `e.g.` is a little awkward. Perhaps something to the effect of: "...functionality exists in usethis, for example,..."

jimhester commented on content/articles/2018-02-usethis-1-3-0.Rmd:
```diff
@@ -0,0 +1,106 @@
+devtools will continue to exist as a meta-package, exposing functionality from lower-level, domain-specific packages. In the long-run, a useR should attach devtools, via `library("devtools")`, and then call devtools functions in their daily workflow, as usual. There are two reasons you might need to know about our internal refactoring:
+
+  * The Awkward Adolescent Phase. It is impossible to choreograph the release of a suite of packages on CRAN. Therefore, there may be periods when functionality exists in, e.g., usethis, that is not yet exposed by devtools. During such times, you would access those specific functions directly through usethis. This implies you should attach devtools first and then any of its constituent packages,  ...
+  * Programming and Development. If you are using devtools functions programmatically, i.e. in production scripts or in another package, it is recommended to do so via the function's primary package home and not through devtools.
```
I wonder if we should be stronger about this (remove 'it is recommend to') and give some justification about the reasons.

1. lighter dependencies for your package (you only need features you use)
2. easier to release new versions of devtools (fewer reverse dependencies)
3. finer grained versioning (can depend on pkgbuild 1.2.3 and pkgload 4.5.6 even if devtools uses pkgbuild 4.5.6 and pkgload 4.5.6)

batpigandme replied:
Add an example here, perhaps?

hadley replied:
And maybe an analogy to tidyverse (and reinforce that's it's bad to this for tidyverse too)

Maybe `use_package()` should warn if you try and add a dependency to devtools?

batpigandme commented on content/articles/2018-02-usethis-1-3-0.Rmd:
```diff
@@ -0,0 +1,106 @@
+
+## Community-specific conventions
+
+The development conventions for [tidyverse packages](https://github.com/tidyverse/) are gradually being codified into a set of [`use_tidy_*()`](http://usethis.r-lib.org/reference/tidyverse.html) helper functions.
```
Do we want to be a bit more explicit about what is/is not appropriate for external use? For example, I think the contributing guidelines use tidyverse-specific language, which would be thoroughly confusing should anyone else unwittingly keep that language in their own CONTRIBUTING.md.

jennybc replied:
Current version clarifies this.

hadley commented on content/articles/2018-02-usethis-1-3-0.Rmd:
```diff
@@ -0,0 +1,106 @@
+
+## usethis vs. devtools
+
+[devtools](https://github.com/r-lib/devtools#readme) is an R package that helps you create, develop, and install R packages. By 2017, it was clear that devtools would be easier to maintain if it was split into a few smaller and more focused packages. usethis is the first of these to be released publicly on CRAN. usethis is the home for functions related to package and project setup.
```
Also because devtools was trying to serve two radically opposing aims: a battery-included install so you could easily develop packages locally, and a minimal-dependency installer and checker (e.g. for travis)

jennybc commented:
With @hadley's blessing, I'm going to move the entire devools story into it's own post. More soon.

jennybc commented:
* removed the section re: devtools breakup (PR for that post coming tomorrow or Monday)
  * added section about usethis's value beyond direct utility of its functions. Too touch feely or OK?
  * populated acknowledgements section

I consider this done (other than replacing placeholder link to the not-yet-written devtools post).
